test/replication-luatest/election_fencing_test.lua

```diff
@@ -0,0 +1,293 @@
+local luatest = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+local g_async = luatest.group('fencing_async', {
+    {election_mode = 'manual'}, {election_mode = 'candidate'}})
+local g_sync = luatest.group('fencing_sync')
+
+local SHORT_TIMEOUT = 0.1
+local LONG_TIMEOUT = 1000
+local DEATH_TIMEOUT = 2 * SHORT_TIMEOUT
+
+local function promote(server)
+    luatest.helpers.retrying({}, function()
+        luatest.assert(server:exec(function()
+            pcall(box.ctl.promote)
+            if box.info.election.state ~= 'leader' then
+                return false
+            end
+            if box.info.synchro.queue.owner ~= box.info.id then
+                return false
+            end
+            return true
+        end))
+    end)
+    server:exec(function() box.ctl.wait_rw() end)
+end
+
+local function wait_sync(leader, servers)
+    local vclock = leader:get_vclock()
+    vclock[0] = nil
+    for _, server in ipairs(servers) do
+        server:wait_vclock(vclock)
+    end
+end
+
+local function wait_disconnected(node_1, node_2)
+    luatest.helpers.retrying({}, function()
+        luatest.assert(node_1:exec(function(i)
+            return (box.info.replication[i].downstream.status == 'stopped' or
+                    box.info.replication[i].upstream == nil or
+                    box.info.replication[i].upstream.status == 'disconnected')
+        end, {node_2:instance_id()}))
+    end)
+end
+
+local function wait_connected(node_1, node_2)
+    luatest.helpers.retrying({}, function()
+        luatest.assert(node_1:exec(function(i)
+            return (box.info.replication[i].downstream.status == 'follow' and
+                    box.info.replication[i].upstream.status == 'follow')
+        end, {node_2:instance_id()}))
+    end)
+end
+
+local function test_rw(server)
+    return server:exec(function()
+        return pcall(box.space.test.replace, box.space.test, {1})
+    end)
+end
+
+local function box_cfg_update(servers, cfg)
+    for _, server in ipairs(servers) do
+        server:exec(function(cfg) box.cfg(cfg) end, {cfg})
+    end
+end
+
+local function start(g)
+    g.box_cfg = {
+        election_mode = 'manual',
+        election_timeout = SHORT_TIMEOUT,
+        replication = {
+            server.build_instance_uri('server_1'),
+            server.build_instance_uri('server_2'),
+            server.build_instance_uri('server_3'),
+        },
+        replication_connect_quorum = 1,
+        replication_synchro_quorum = 2,
+        replication_synchro_timeout = SHORT_TIMEOUT,
```

Comment:
Let's bump `replication_synchro_timeout` to 4 * SHORT_TIMEOUT, at least. This will reduce potential flakiness on slow servers and will be consistent with replication_disconnect_timeout() = 4 * replication_timeout.

Reply:
Nevermind, I see why you use such a small timeout. This helps to test that nothing is rolled back. And keeps the test time small.